test/e2e/alertmanager_test.go

```diff
@@ -242,6 +245,48 @@ func testAMMeshInitialization(t *testing.T) {
 	}
 }
 
+// testAMMeshAfterRollingUpdate tests whether all Alertmanager instances join
+// the cluster after a rolling update, even though DNS records will probably be
+// outdated at startup time. See
+// https://github.com/prometheus/alertmanager/pull/1428 for more details.
+func testAMMeshAfterRollingUpdate(t *testing.T) {
+	var err error
+
+	t.Parallel()
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	amClusterSize := 3
+
+	alertmanager := framework.MakeBasicAlertmanager("test", int32(amClusterSize))
```

Comment:
:/ it's too bad AM can only handle 2^32 nodes in a cluster

Reply:
Even worse, int32 is signed :(

https://golang.org/pkg/builtin/#int32

Reply:
ugh, this won't do. we really need uint64 for any real world scenarios

Reply:
https://media.giphy.com/media/may13i5UI9aFO/giphy.gif